core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -2113,7 +2113,12 @@ class KafkaZkClient private[zk] (zooKeeperClient: ZooKeeperClient, isSecure: Boo
   }
 
   private class CheckedEphemeral(path: String, data: Array[Byte]) extends Logging {
+    private var attempt = 0
+    private val maxAttempt = 5
+    private val backoffMs = 1000
```

Comment:
I wonder if it would make sense to have the interval between retries be defined as: `(zkSessionTimeout + margin) / maxAttempts`? Where `margin` could be a small fixed delta, e.g. one second, and `zkSessionTimeout` is taken from the Kafka configuration for `"zookeeper.session.timeout.ms"`. WDYT?

Reply:
This makes sense, sure!